### PR TITLE
added new files

### PR DESCRIPTION
**Pull request by YaroslavBorysko**

### CODE REVIEW

File: tasks/task_178d.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    """
+    if not isinstance(n, int):
+        raise TypeError("Argument should be integer!")
+    if n <= 0:
+        raise ValueError("Argument is not natural number!")
+    c = 0
+    for elem in range(n):
+        if arr[elem] > 2**elem and arr[elem] < fac(elem):
+            c += 1
+    return c
```
Comment (therobotisnotatoy) [inline tasks/task_178d.py]: Empty line required at the end of file

File: tasks/task_178d.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+"""Module with task 178d"""
+from math import factorial as fac
```
Comment (Misha86) [inline tasks/task_178d.py:3]: need empty string after module docstring

File: tasks/task_178d.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+"""Module with task 178d"""
```
Comment (therobotisnotatoy) [inline tasks/task_178d.py:1]: Empty line after module doctoring

File: tasks/task_178d.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+from math import factorial as fac
+
+
+def task178d(n: int, arr: list) -> int:
```
Comment (Misha86) [inline tasks/task_178d.py]: change format docstring
"""Docstring.......
...........
"""

File: tests/test_task_182.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+"""Module with tests for task 182"""
+from unittest import TestCase
+from tasks.tasks_182 import task_182
+
+
+class TestTask182(TestCase):
+    """
```
Comment (Misha86) [inline tests/test_task_182.py]: change format docstring
"""Docstring.......
...........
"""

File: tests/test_task_182.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    Tests for task 182
+    """
+    def test_assert_equal(self):
+        """
```
Comment (Misha86) [inline tests/test_task_182.py]: change format docstring
"""Docstring.......
...........
"""

File: tests/test_task_182.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        Test with positive numbers in sequence
+        """
+        test_n = 5
+        test_list = [6, 35, 40, 97, 21]
+        self.assertEqual(task_182(test_n, test_list), (1, 40))
+
+    def test_assert_raises(self):
+        """
```
Comment (Misha86) [inline tests/test_task_182.py]: change format docstring
"""Docstring.......
...........
"""

File: tests/test_task_182.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        Test with string in sequence
+        """
+        self.assertRaises(AssertionError, task_182, "3", "[2, 3, 4]")
+
+    def test_with_negative_arr(self):
+        """
```
Comment (Misha86) [inline tests/test_task_182.py]: change format docstring
"""Docstring.......
...........
"""

File: tests/test_task_182.py [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+        Test with negative numbers in sequence
+        """
+        test_n = 1
+        test_list = [-6]
+        self.assertEqual(task_182(test_n, test_list), (0, 0))
+
+    def test_with_zero(self):
+        """
```
Comment (Misha86) [inline tests/test_task_182.py]: change format docstring
"""Docstring.......
...........
"""

File: tests/test_task_323.py [hunk 10]
```diff
@@ -0,0 +1,30 @@
+"""Module with tests for task 323"""
+from unittest import TestCase
+from tasks.tasks_323 import task_323
+
+class TestTask323(TestCase):
+    """
```
Comment (Misha86) [inline tests/test_task_323.py]: change format docstring
"""Docstring.......
...........
"""

File: tasks/task_178g.py [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+    """
+    if not isinstance(n, int):
+        raise TypeError("Argument should be integer!")
+    if n <= 0:
+        raise ValueError("Argument is not natural number!")
+    c = 0
+    for elem in range(1, n - 1):
+        if arr[elem] < (arr[elem + 1] + arr[elem - 1]) / 2:
+            c += 1
+    return c
```
Comment (therobotisnotatoy) [inline tasks/task_178g.py]: Is it ok to use a single character as a variable?

File: tasks/task_555.py [hunk 12]
```diff
@@ -0,0 +1,16 @@
+"""Module with task 555"""
```
Comment (Misha86) [inline tasks/task_555.py:1]: need empty string after module docstring

File: tasks/task_555.py [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+    Build pascal's triangle with n floors
+    """
+    if not isinstance(n, int):
+        raise TypeError("Argument should be integer!")
+    if n <= 0:
+        raise ValueError("Argument is not natural number!")
+
+    return [[int((fac(n) / (fac(k) * fac(n - k)))) for k in range(0, n + 1)] for n in range(0, n)]
+
+
```
Comment (therobotisnotatoy) [inline tasks/task_555.py]: Redundant line

File: tasks/task_178g.py [hunk 14]
```diff
@@ -0,0 +1,17 @@
+"""Module with task 178d"""
+
+
+def task178g(n: int, arr: list) -> int:
+    """
```
Comment (Misha86) [inline tasks/task_178g.py]: change format docstring
"""Docstring.......
...........
"""

File: tasks/task_555.py [hunk 15]
```diff
@@ -0,0 +1,16 @@
+"""Module with task 555"""
+from math import factorial as fac
+
+
+def task555(n: int) -> list:
+    """
```
Comment (Misha86) [inline tasks/task_555.py]: change format docstring
"""Docstring.......
...........
"""

File: tests/test_task_182.py [hunk 16]
```diff
@@ -0,0 +1,38 @@
+"""Module with tests for task 182"""
```
Comment (Misha86) [inline tests/test_task_182.py:1]: need empty string

File: tests/test_task_182.py [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+from unittest import TestCase
+from tasks.tasks_182 import task_182
```
Comment (therobotisnotatoy) [inline tests/test_task_182.py:5]: Split different groups of import by empty line

File: tests/test_task_323.py [hunk 18]
```diff
@@ -0,0 +1,30 @@
+"""Module with tests for task 323"""
```
Comment (Misha86) [inline tests/test_task_323.py:1]: need empty string

File: tests/test_task_323.py [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+    def test_assert_equal(self):
+        """
+        Test with positive number
+        """
+        test_n = 6
+        test_res = ['5 & 4', '5 & 3', '5 & 2', '5 & 1', '4 & 3', '4 & 1', '3 & 2', '3 & 1', '2 & 1', '1 & 1']
+        self.assertEqual(task_323(test_n), test_res)
+
+    def test_assert_raises(self):
+        """
```
Comment (Misha86) [inline tests/test_task_323.py]: change format docstring
"""Docstring.......
...........
"""

File: tests/test_task_323.py [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+        Test with string
+        """
+        test_n = "9"
+        self.assertRaises(AssertionError, task_323, test_n)
+
+    def test_negative(self):
+        """
```
Comment (Misha86) [inline tests/test_task_323.py]: change format docstring
"""Docstring.......
...........
"""

File: tests/test_task_560.py [hunk 21]
```diff
@@ -0,0 +1,22 @@
+"""Module with tests for task 560"""
+from unittest import TestCase
+from tasks.task_560 import task_560
+
+
+class TestTask560(TestCase):
```
Comment (Misha86) [inline tests/test_task_560.py:8]: change format docstring
"""Docstring.......
...........
"""

File: tests/test_task_560.py [hunk 22]
```diff
@@ -0,0 +1,22 @@
+"""Module with tests for task 560"""
```
Comment (Misha86) [inline tests/test_task_560.py:1]: need empty string

File: tests/test_task_323.py [hunk 23]
```diff
@@ -0,0 +1,30 @@
+"""Module with tests for task 323"""
+from unittest import TestCase
+from tasks.tasks_323 import task_323
+
+class TestTask323(TestCase):
+    """
+    Tests for task 323
+    """
```
Comment (Misha86) [inline tests/test_task_323.py]: need empty string

File: tests/test_task_560.py [hunk 24]
```diff
@@ -0,0 +1,22 @@
...
+from unittest import TestCase
+from tasks.task_560 import task_560
+
+
+class TestTask560(TestCase):
+    """
+    Tests for task 560
+    """
```
Comment (Misha86) [inline tests/test_task_560.py]: need empty string

File: tests/test_task_560.py [hunk 25]
```diff
@@ -0,0 +1,22 @@
...
+    def test_valid(self):
+        """
```
Comment (Misha86) [inline tests/test_task_560.py]: change format docstring
"""Docstring.......
...........
"""

File: tests/test_task_560.py [hunk 26]
```diff
@@ -0,0 +1,22 @@
...
+        Test with valid result
+        """
+        test_res = ['220 284']
+        self.assertEqual(task_560(), test_res)
+
+    def test_not_valid(self):
+        """
```
Comment (Misha86) [inline tests/test_task_560.py]: change format docstring
"""Docstring.......
...........
"""

**Review by Misha86 — COMMENTED** [review]
change format docstring
"""Docstring.......
...........
"""
add empty strings after test`s docstring

File: tasks/task_178d.py [hunk 27]
```diff
@@ -0,0 +1,17 @@
+"""Module with task 178d"""
+
+from math import factorial as fac
+
+from tasks.utils import is_natural_number
+
+
+def task178d(num: int, arr: list) -> int:
+    """ Return number that satisfy a following condition:
```
Comment (morento101) [inline tasks/task_178d.py:8]: Stick to the task naming convention task_<task_number>

File: tests/test_task_182.py [hunk 28]
```diff
@@ -0,0 +1,30 @@
+"""Module with tests for task 182"""
+
+from unittest import TestCase
+
+from tasks.tasks_182 import task_182
+
+
+class TestTask182(TestCase):
+    """ Tests for task 182 """
```
Comment (morento101) [inline tests/test_task_182.py:9]: add whitespace after Test class doc str

**Review by morento101 — CHANGES_REQUESTED** [review]
Change function names and add a whitespace after Test class docstring and your branch is ready for merge

File: tests/test_task_323.py [hunk 29]
```diff
@@ -0,0 +1,26 @@
+"""Module with tests for task 323"""
+
+from unittest import TestCase
+
+from tasks.tasks_323 import task_323
+
+class TestTask323(TestCase):
+    """Tests for task 323"""
```
Comment (morento101) [inline tests/test_task_323.py:8]: Whitespace after doc str

File: tests/test_task_560.py [hunk 30]
```diff
@@ -0,0 +1,18 @@
+"""Module with tests for task 560"""
+
+from unitest import TestCase
+
+from tasks.task_560 import task_560
+
+
+class TestTask560(TestCase):
+    """Tests for task 560"""
+    def test_valid(self):
```
Comment (morento101) [inline tests/test_task_560.py:11]: Whitespace after class doc str

**Review by morento101 — APPROVED** [review]
Great job!

**Pull request merged**